addons/l10n_it_edi_imp/models/account_move.py

```diff
@@ -0,0 +1,53 @@
+from collections import defaultdict
+
+from odoo.addons.l10n_it_edi_imp.models.account_payment_methode_line import L10N_IT_PAYMENT_STATE_SELECTION
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_it_payment_method = fields.Selection(
+        selection=L10N_IT_PAYMENT_STATE_SELECTION,
+        compute='_compute_l10n_it_payment_method',
+        store=True,
+        readonly=False,
+    )
+
+    l10n_it_document_type = fields.Many2one(
+        comodel_name='l10n_it.document.type',
+        compute='_compute_l10n_it_document_type',
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends('line_ids.matching_number', 'payment_state')
+    def _compute_l10n_it_payment_method(self):
+        line_matching_numbers = {line.matching_number for move in self for line in move.line_ids if line.matching_number}
+        all_lines_with_matching_number = self.env['account.move.line'].search([('matching_number', 'in', list(line_matching_numbers))])
+        move_line_per_matching_number = defaultdict(self.env['account.move.line'].browse)
+        # We create a dict with all the matching number and the lines associated to it
+        for line in all_lines_with_matching_number:
+            move_line_per_matching_number[line.matching_number] += line
+
+        for move in self:
+            matching_numbers = move.line_ids.filtered('matching_number').mapped('matching_number')
+            if matching_numbers and move.payment_state in {'in_payment', 'partial'}:
+                matching_lines = move_line_per_matching_number.get(matching_numbers[0])
+                # We get the first one by default if there is multiple lines
+                payment_method_line = matching_lines.payment_id.payment_method_line_id[0]
+                move.l10n_it_payment_method = payment_method_line.l10n_it_payment_method
+            else:
+                move.l10n_it_payment_method = move.payment_id.payment_method_line_id.l10n_it_payment_method or move.l10n_it_payment_method or 'MP05'
+
+    @api.depends('line_ids')
+    def _compute_l10n_it_document_type(self):
+        for move in self:
+            move.l10n_it_document_type = move._l10n_it_edi_get_document_type()
```

Comment:
I don't think a simple if not l10n_it_document_type would work cause if the person modify the lines, a document type will be set, then if he modify again the lines again it will not be recomputed and then could be wrong 🤔
Don't know the best approach, i will check with benj tomorrow 👍